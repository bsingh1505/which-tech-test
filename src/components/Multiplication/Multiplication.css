.Container {
  display: flex;
  flex-flow: row wrap;
  text-align: center;
  padding: 0;
  margin: 0;
  list-style: none;
  max-width: 1280px;
  align-content: center;
}
.Item {
  background: tomato;
  width: 400px;
  height: 250px;
  margin: 10px;
  color: white;
  font-weight: bold;
  font-size: 5em;
  text-align: center;
  line-height: 250px;
}
.Item-selected {
  background: green !important;
}
@media screen and (min-width: 1600px) {
  .Container {
    border: 10px;
    border-style: dotted;
  }
}
