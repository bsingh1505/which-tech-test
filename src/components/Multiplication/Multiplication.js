import React, { useState } from "react";
import "./Multiplication.css";

const MULTIPLICATIONS_UPTO = 144;

const Multiplication = ({ multiplicationsUpto = MULTIPLICATIONS_UPTO }) => {
  const numberToArray = Array.from(Array(multiplicationsUpto).keys());
  const [selectedItem, setSelectedItem] = useState(1);

  const handleItemClick = (clickedItem) => {
    setSelectedItem(clickedItem);
  };

  const applyClass = (currentNumber) => {
    const isMultiple = currentNumber % selectedItem === 0;
    return isMultiple ? `Item Item-selected` : `Item`;
  };

  return (
    <>
      <ul className="Container">
        {numberToArray.map((item) => {
          const currentNumber = item + 1;
          return (
            <li
              key={`key-${currentNumber}`}
              data-testid={"list-item"}
              className={applyClass(currentNumber)}
              onClick={() => handleItemClick(currentNumber)}
            >
              {currentNumber}
            </li>
          );
        })}
      </ul>
    </>
  );
};

export default Multiplication;
