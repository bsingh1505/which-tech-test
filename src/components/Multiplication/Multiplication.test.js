import { render, fireEvent } from "@testing-library/react";
import Multiplication from "./Multiplication";

const setup = (overrideProps) => {
  const props = {
    multiplicationsUpto: 10,
    ...overrideProps,
  };
  const { container, getByText, getAllByTestId } = render(<Multiplication {...props} />);
  return {
    container,
    props,
    getByText,
    getAllByTestId,
  };
};

describe("Multiplcation component", () => {
  test(":: should render component", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  test(":: one click of item it should match snapshot and proper class assigned", () => {
    const { getByText } = setup();
    const item = getByText("3");
    const item4 = getByText("4");
    const item6 = getByText("6");
    fireEvent.click(item);
    expect(item).toHaveClass("Item-selected");
    expect(item4).not.toHaveClass("Item-selected");
    expect(item6).toHaveClass("Item-selected");
  });

  test(":: should render all the numbers", () => {
    const { getAllByTestId } = setup();
    const listItems = getAllByTestId("list-item");
    expect(listItems).toHaveLength(10);
  });

  test(":: should render expected numbers of item", () => {
    const { getAllByTestId } = setup({ multiplicationsUpto: 5 });
    const listItems = getAllByTestId("list-item");
    expect(listItems).toHaveLength(5);
  });
});
