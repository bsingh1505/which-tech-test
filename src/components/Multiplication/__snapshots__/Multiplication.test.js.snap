// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiplcation component :: should render component 1`] = `
<div>
  <ul
    class="Container"
  >
    <li
      class="Item Item-selected"
      data-testid="list-item"
    >
      1
    </li>
    <li
      class="Item Item-selected"
      data-testid="list-item"
    >
      2
    </li>
    <li
      class="Item Item-selected"
      data-testid="list-item"
    >
      3
    </li>
    <li
      class="Item Item-selected"
      data-testid="list-item"
    >
      4
    </li>
    <li
      class="Item Item-selected"
      data-testid="list-item"
    >
      5
    </li>
    <li
      class="Item Item-selected"
      data-testid="list-item"
    >
      6
    </li>
    <li
      class="Item Item-selected"
      data-testid="list-item"
    >
      7
    </li>
    <li
      class="Item Item-selected"
      data-testid="list-item"
    >
      8
    </li>
    <li
      class="Item Item-selected"
      data-testid="list-item"
    >
      9
    </li>
    <li
      class="Item Item-selected"
      data-testid="list-item"
    >
      10
    </li>
  </ul>
</div>
`;
